components/suggestion/at_mention_provider/at_mention_provider.jsx

```diff
@@ -15,10 +13,11 @@ import Provider from '../provider.jsx';
 import AtMentionSuggestion from './at_mention_suggestion.jsx';
 
 const profilesInChannelOptions = {active: true};
+const regexForAtMention = /(?:^|\W)@([\p{L}\d_. -]*)$/ui;
```

Comment:
I don't know enough about regex or xregexp to be sure if this is a intended change, but after the `\d`, I see two main differences with what was being passed to xregexp:
- We are missing a `\` (or `\\`)
- The order is different

Reply:
- `\\` is need for escaping when we have regex as string
- yes the order doesn't matter for this character class `[\p{L}\d_. -]` because it will try to match a single character in that set. Reason for putting `-` in the end was `\d-` tries to create a range from `\d` to `_` and it becomes invalid regex. This wasnt before as now with addition of `u` flag regex becomes a bit strict (read pain)